syntax/verilog_systemverilog.vim

```diff
@@ -150,111 +150,63 @@ endif
-else
-    syn keyword verilogStatement class endclass
-endif
-if index(s:verilog_syntax_fold, "interface") >= 0 || index(s:verilog_syntax_fold, "all") >= 0
-    syn region  verilogFold
-        \ matchgroup=verilogStatement
-        \ start="\<interface\>\(\s\+class\)\@!"
-        \ end="\<endinterface\>"
-        \ transparent
-        \ keepend
-        \ fold
-else
-    syn keyword verilogStatement interface endinterface
-endif
-if index(s:verilog_syntax_fold, "clocking") >= 0 || index(s:verilog_syntax_fold, "all") >= 0
-    syn region  verilogFold
-        \ matchgroup=verilogStatement
-        \ start="\<clocking\>"
-        \ end="\<endclocking\>"
-        \ transparent
-        \ keepend
-        \ fold
-else
-    syn keyword verilogStatement clocking endclocking
-endif
-if index(s:verilog_syntax_fold, "covergroup") >= 0 || index(s:verilog_syntax_fold, "all") >= 0
-    syn region  verilogFold
-        \ matchgroup=verilogStatement
-        \ start="\<covergroup\>"
-        \ end="\<endgroup\>"
-        \ transparent
-        \ keepend
-        \ fold
-else
-    syn keyword verilogStatement  covergroup endgroup
-endif
-if index(s:verilog_syntax_fold, "sequence") >= 0 || index(s:verilog_syntax_fold, "all") >= 0
-    syn region  verilogFold
-        \ matchgroup=verilogStatement
-        \ start="\<sequence\>"
-        \ end="\<endsequence\>"
-        \ transparent
-        \ keepend
-        \ fold
-else
-    syn keyword verilogStatement  sequence endsequence
-endif
-if index(s:verilog_syntax_fold, "property") >= 0 || index(s:verilog_syntax_fold, "all") >= 0
-    syn region  verilogFold
-        \ matchgroup=verilogStatement
-        \ start="\<property\>"
-        \ end="\<endproperty\>"
-        \ transparent
-        \ keepend
-        \ fold
-else
-    syn keyword verilogStatement  property endproperty
-endif
-if index(s:verilog_syntax_fold, "specify") >= 0 || index(s:verilog_syntax_fold, "all") >= 0
-    syn region  verilogFold
-        \ matchgroup=verilogStatement
-        \ start="\<specify\>"
-        \ end="\<endspecify\>"
-        \ transparent
-        \ keepend
-        \ fold
-else
-    syn keyword verilogStatement  specify endspecify
-endif
-if index(s:verilog_syntax_fold, "block_nested") >= 0 || index(s:verilog_syntax_fold, "block_named") >= 0
+for name in ['class', 'clocking', 'covergroup', 'function', 'interface',
+    \ 'property', 'sequence', 'specify', 'task', ]
+
+    if name == 'task' || name == 'function'
+        let region_start=verilog_function_task_dequalifier.'\@<!\<'.name.'\>'
+    elseif name == 'class'
+        let region_start='\(typedef\s\+\)\@<!\<\(interface\s\+\)\?class\>'
+    elseif name == 'interface'
+        let region_start='\<interface\>\(\s\+class\)\@!'
+    else
+        let region_start='\<'.name.'\>'
+    endif
+
+    if name == 'covergroup'
+        let region_end = '\<endgroup\>'
+    else
+        let region_end = '\<end'.name.'\>'
+    endif
+
+    let verilog_syn_region_cmd_{name} =
```

Comment:
Funny fact: I learned about `{var}` while looking at code from QuickFixSigns 2 or 3 days before I started this review!